### PR TITLE
Propagate YARN to attention scale

### PR DESCRIPTION
**Pull request by sempervictus**
## Problem

For YARN-scaled models (e.g., 12X context extension), the attention
softmax scale was not being adjusted for the extended context window.
This caused attention weights to be too sharp, leading to:
- Poor gradient flow during inference
- Repetitive/looping output as the model attends to incorrect positions
- Degraded generation quality for long contexts

The YARN scaling factor (mscale) compensates for the extended context
by scaling the attention logits before softmax. Without this scaling,
positions beyond the original training context length receive incorrect
attention weights.

## Root Cause

MLA models (DeepSeek) correctly applied YARN scaling in MlaAttention::new()
via `sm_scale *= mscale * mscale`. However, non-MLA models (LLaMA, Qwen3,
etc.) used Attention::new() which passed `None` for attention_scale,
resulting in the default `1.0 / sqrt(head_dim)` without YARN scaling.

## Solution

Applied YARN scaling computation to Attention::new_with_options() for
non-MLA models. When rope_scaling.type == "yarn", the attention_scale
is now computed as:

    base_scale = attention_scale.unwrap_or(1.0 / sqrt(head_dim))
    mscale = 0.1 * mscale_all_dim * log(factor) + 1.0
    sm_scale = base_scale * mscale * mscale

This matches the existing MLA implementation and ensures consistent
attention scaling across all model architectures.

## Data Flow

vllm.rs Config.rope_scaling (from model config or CLI --yarn-scaling-factor)
  -> Attention::new_with_options() or MlaAttention::new()
  -> sm_scale *= mscale * mscale for YARN models
  -> PagedAttention::new() or MLA attention
  -> CUDA kernel uses scaled sm_scale for softmax

### CODE REVIEW

**Comment by sempervictus** [comment]
Doesn't outright break stuff at long-context... :smile: 

<details>

image

</details>

_without_ this PR i can somewhat reproducibly get this kind of effect on the NVFP4 Q3Next/3.5/3.6 models on a single SM121:

<details>
image

</details>

_just_ after we cross the original training threshold of max embedding positions ... which seemed very suspect to me and how i got to this PR in the first place (well, second place :-) )

**Comment by guoqingbao** [comment]
> Doesn't outright break stuff at long-context... 😄

But I don't find any uses of `attention_scale` in this PR, the scaling rope already done the yarn scaling.
